oauth2_http/java/com/google/auth/oauth2/ServiceAccountCredentials.java

```diff
@@ -515,4 +480,127 @@ String createAssertion(JsonFactory jsonFactory, long currentTime) throws IOExcep
+    private PrivateKey privateKey;
+    private String privateKeyId;
+    private String serviceAccountUser;
+    private String projectId;
+    private URI tokenServerUri;
+    private Collection<String> scopes;
+    private HttpTransportFactory transportFactory;
+
+    public ServiceAccountCredentials build() {
+      return new ServiceAccountCredentials(
+          clientId, clientEmail, privateKey, privateKeyId, scopes,
+          transportFactory, tokenServerUri, serviceAccountUser, projectId);
+    }
+
+    public Builder setClientId(String clientId) {
+      this.clientId = clientId;
+      return this;
+    }
+
+    public Builder setClientEmail(String clientEmail) {
+      this.clientEmail = clientEmail;
+      return this;
+    }
+
+    public Builder setPrivateKey(PrivateKey privateKey) {
+      this.privateKey = privateKey;
+      return this;
+    }
+
+    public Builder setPrivateKeyId(String privateKeyId) {
+      this.privateKeyId = privateKeyId;
+      return this;
+    }
+
+    public Builder setScopes(Collection<String> scopes) {
+      this.scopes = scopes;
+      return this;
+    }
+
+    public Builder setServiceAccountUser(String serviceAccountUser) {
+      this.serviceAccountUser = serviceAccountUser;
+      return this;
+    }
+
+    public Builder setProjectId(String projectId) {
+      this.projectId = projectId;
+      return this;
+    }
+
+    public Builder setTokenServerUri(URI tokenServerUri) {
+      this.tokenServerUri = tokenServerUri;
+      return this;
+    }
+
+
+    public Builder setHttpTransportFactory(HttpTransportFactory transportFactory) {
+      this.transportFactory = transportFactory;
+      return this;
+    }
+
+    public String getClientId() {
+      return clientId;
+    }
+
+    public String getClientEmail() {
+      return clientEmail;
+    }
+
+    public PrivateKey getPrivateKey() {
+      return privateKey;
+    }
+
+    public String getPrivateKeyId() {
+      return privateKeyId;
+    }
+
+    public Collection<String> getScopes() {
+      return scopes;
+    }
+
+    public String getServiceAccountUser() {
+      return serviceAccountUser;
+    }
+
+    public String getProjectId() {
+      return projectId;
+    }
+
+    public URI getTokenServerUri() {
+      return tokenServerUri;
+    }
+
+    public HttpTransportFactory getHttpTransportFactory() {
+      return transportFactory;
+    }
+
+    protected Builder() {}
+
+    protected Builder(ServiceAccountCredentials credentials) {
+      this.clientId = credentials.clientId;
+      this.clientEmail = credentials.clientEmail;
+      this.privateKey = credentials.privateKey;
+      this.privateKeyId = credentials.privateKeyId;
+      this.scopes = credentials.scopes;
+      this.transportFactory = credentials.transportFactory;
+      this.tokenServerUri = credentials.tokenServerUri;
+      this.serviceAccountUser = credentials.serviceAccountUser;
+      this.projectId = credentials.projectId;
+    }
```

Comment:
Your ordering here is not the convention we normally follow. Use this order:

1. constructors
2. get/set methods
3. build()

Reply:
Done